### PR TITLE
Updating Wolfi with Dockerfiles guide

### PR DESCRIPTION
**Pull request by erikaheidi**
This PR brings some updates to the new Wolfi with Dockerfiles guides, based on feedback from @amouat and @ltagliaferri .

It also fixes a typo in the melange tutorial tags definition.

Preview: https://deploy-preview-548--ornate-narwhal-088216.netlify.app/open-source/wolfi/wolfi-with-dockerfiles/

### CODE REVIEW

File: content/open-source/wolfi/wolfi-with-dockerfiles.md [hunk 1]
```diff
@@ -110,10 +97,12 @@ Copy the following content to it:
 ```
 FROM cgr.dev/chainguard/wolfi-base
```
Comment (amouat) [inline content/open-source/wolfi/wolfi-with-dockerfiles.md:98]: I'm fine with this, I just wonder if using python:$version-dev would be simpler?

File: content/open-source/wolfi/wolfi-with-dockerfiles.md [hunk 2]
```diff
@@ -175,10 +167,10 @@ RUN  pip3 install -r requirements.txt --user
 
 FROM cgr.dev/chainguard/python:latest
```
Comment (amouat) [inline content/open-source/wolfi/wolfi-with-dockerfiles.md:168]: think you can do `python:$version` to avoid compatibility problems.

**Review by amouat — APPROVED** [review]

File: content/open-source/wolfi/wolfi-with-dockerfiles.md [hunk 3]
```diff
@@ -110,10 +97,12 @@ Copy the following content to it:
...
 FROM cgr.dev/chainguard/wolfi-base
 
+# Use this variable to set up Python version
+ARG version=3.11
+
 RUN adduser -D nonroot
 WORKDIR /app
-
-RUN apk add python-3.11 py3.11-pip && \
+RUN apk add python-$version py$version-pip && \
```
Comment (amouat) [inline content/open-source/wolfi/wolfi-with-dockerfiles.md:105]: you should be able to do `py${version}-pip` which is a bit easier to read.

**Review by ltagliaferri — APPROVED** [review]
LGTM after you address @amouat's edits, thank you @erikaheidi!

**Comment by erikaheidi** [comment]
> Great!
> 
> Small things:
> 
>     * You can use `${version}` to make it clearer where the variable starts and stops
> 
>     * There's still going to be a problem when `latest` moves version I _think_
>       
>       * can't we use `python:3.11` instead? You can even do this I think:
> 
> 
> ```
> ARG version=3.11
> FROM cgr.dev/chainguard/python:$version
> ```

I tried the `FROM` line with a variable but it doesn't seem to work, maybe the variable is not in that context? 

```
Step 9/14 : FROM cgr.dev/chainguard/python:$version
invalid reference format

```

**Comment by amouat** [comment]
Interesting, let me have a look...

**Comment by amouat** [comment]
OK, it's an interesting one. If you move the `ARG version=3.11` line to be the first line in the file (before the first FROM) it should work.

**Comment by erikaheidi** [comment]
> 

That didn't work here - now the variable is not resolved:
```
fetch https://packages.wolfi.dev/os/x86_64/APKINDEX.tar.gz
ERROR: unable to select packages:
  py-pip (no such package):
    required by: world[py-pip]
  python- (no such package):
    required by: world[python-]
```

I have changed it to use the 3.11 tag instead of latest, let me know if that works.

**Comment by amouat** [comment]
Sure, that's fine.

Can you send me the Dockerfile that isn't working for you?

**Comment by erikaheidi** [comment]
> Sure, that's fine.
> 
> Can you send me the Dockerfile that isn't working for you?

The Dockerfile only worked with the VAR added to three locations (at the top, and on each stage). I am not sure this is the best way because of the duplication. So I'll be merging these updates now to incorporate them asap on the live piece, but we can always update the Dockerfile.

**Pull request merged**